src/Utilities/Functional.hpp

```diff
@@ -0,0 +1,159 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+#include <tuple>
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Higher order function objects similar to `std::plus`, etc.
+ */
+namespace funcl {
+/// \cond
+template <size_t Arity>
+struct Functional {
+  static constexpr size_t arity = Arity;
```

Comment:
arity needs to be re-defined in every operator so arbitrary chaining works. For instance, this accomplishes Plus<Plus<>> but not Plus<Plus<Plus<>>>. maybe also add a test for one deeper iteration in the chain?

Reply:
Yea, we need to keep track of the arity. I added this and a test for it. Good catch!